server/api/user/update/controller.test.js

```diff
@@ -0,0 +1,355 @@
+      error: {
+        code: 10004,
+        message: `You need to verify ${badEmails[0].address} before marking it as primary`,
+      },
+    });
+  });
+
+  test('returns error when trying to add an org to the user', async () => {
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        orgs: ['5b2d5dd0cd86b77258e16d39'], //random id
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: false,
+      error: {
+        code: 400,
+        message: 'Invalid request body',
+      },
+    });
+  });
+
+  test('updates user with new image if picture url is provided', async () => {
+    const newPictureURL = 'https://www.acme.com/pictures/v2/coyote.png';
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        picture: newPictureURL,
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: true,
+      user: {
+        id: user.id,
+        picture: newPictureURL,
+        fullName: user.fullName,
+        username: user.username,
+        emails: expect.any(Array),
+        orgs: expect.any(Array),
+      },
+    });
+  });
+  test('removes image from user if picture property is null', async () => {
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        picture: null,
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: true,
+      user: {
+        id: user.id,
+        picture: null,
+        fullName: user.fullName,
+        username: user.username,
+        emails: expect.any(Array),
+        orgs: expect.any(Array),
+      },
+    });
+  });
+
+  test('updates user and returns expected response', async () => {
+    const newPictureURL = 'https://www.acme.com/pictures/v2/coyote.png';
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        fullName: 'new Coyotee',
+        username: 'alibaba',
+        emails: [{
+          address: 'coyote@acme.com',
+          isPrimary: true,
+        }, {
+          address: 'coyote-new@acme.com',
+        }],
+        picture: newPictureURL,
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: true,
+      user: {
+        id: user.id,
+        fullName: 'new Coyotee',
+        username: 'alibaba',
+        emails: expect.arrayContaining([{
+          address: expect.any(String),
+          isPrimary: expect.any(Boolean),
+        }]),
+        picture: newPictureURL,
+      },
+    });
+  });
+
+  test.only('updates user password', async () => {
```

Comment:
I think we need to ask for the old password in case the requestor is not superadmin.